wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -76,6 +76,34 @@ def retrieve_cfme_appliance_version(template_name):
 """
 
 
+def reconnect(decorator):
+    def decorate(cls):
+        for attr in cls.__dict__:
+            if callable(getattr(cls, attr)) and not attr.startswith('_'):
+                setattr(cls, attr, decorator(getattr(cls, attr)))
+        return cls
+    return decorate
+
+
+def unauthenticated_error_handler(method):
+    """Fixes issue with 401 error by restoring connection.
+        Sometimes connection to openshift api endpoint gets expired and openshift returns 401.
+        As a result tasks in some applications like sprout fail.
+    """
+    def wrap(*args, **kwargs):
```

Comment:
**required** a wrapper should use the functools.wraps decorator if possible

Reply:
fixed